src/coreComponents/physicsSolvers/contact/SolidMechanicsAugmentedLagrangianContact.cpp

```diff
@@ -0,0 +1,1347 @@
+{
+
+  GEOS_MARK_FUNCTION;
+
+  SolidMechanicsLagrangianFEM::applySystemSolution( dofManager,
+                                                    localSolution,
+                                                    scalingFactor,
+                                                    dt,
+                                                    domain );
+
+  dofManager.addVectorToField( localSolution,
+                               solidMechanics::totalBubbleDisplacement::key(),
+                               solidMechanics::totalBubbleDisplacement::key(),
+                               scalingFactor );
+
+  dofManager.addVectorToField( localSolution,
+                               solidMechanics::totalBubbleDisplacement::key(),
+                               solidMechanics::incrementalBubbleDisplacement::key(),
+                               scalingFactor );
+
+
+  // Loop for updating the displacement jump
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const & meshName,
+                                                                MeshLevel & mesh,
+                                                                arrayView1d< string const > const & )
+
+  {
+
+    NodeManager const & nodeManager = mesh.getNodeManager();
+    FaceManager const & faceManager = mesh.getFaceManager();
+
+    string const & dispDofKey = dofManager.getKey( solidMechanics::totalDisplacement::key() );
+    string const & bubbleDofKey = dofManager.getKey( solidMechanics::totalBubbleDisplacement::key() );
+
+    arrayView1d< globalIndex const > const dispDofNumber = nodeManager.getReference< globalIndex_array >( dispDofKey );
+    arrayView1d< globalIndex const > const bubbleDofNumber = faceManager.getReference< globalIndex_array >( bubbleDofKey );
+
+    string const & fractureRegionName = this->getUniqueFractureRegionName();
+
+    CRSMatrix< real64, globalIndex > const voidMatrix;
+    array1d< real64 > const voidRhs;
+
+    forFiniteElementOnFractureSubRegions( meshName, [&] ( string const & finiteElementName,
+                                                          arrayView1d< localIndex const > const & faceElementList )
+    {
+
+      finiteElement::FiniteElementBase & subRegionFE = *(m_faceTypeToFiniteElements[finiteElementName]);
+
+      solidMechanicsALMKernels::ALMJumpUpdateFactory kernelFactory( dispDofNumber,
+                                                                    bubbleDofNumber,
+                                                                    dofManager.rankOffset(),
+                                                                    voidMatrix.toViewConstSizes(),
+                                                                    voidRhs.toView(),
+                                                                    dt,
+                                                                    faceElementList );
+
+      real64 maxTraction = finiteElement::
```

Comment:
you probably can define this not to return a max traction since it's just updating the jump.